server/app/services/rpc_server.rb

```diff
@@ -41,20 +74,13 @@ def handle_request(ws_client, grid_id, message)
       begin
         result = actor.send(method, *message[3])
         send_message(ws_client, [1, msg_id, nil, result])
-        unless actor.alive?
-          error "actor for handler #{handler} did die, removing it from cache"
-          @handlers[grid_id].delete(handler)
-        end
-      rescue Celluloid::DeadActorError
-        error "actor for handler #{handler} is dead, removing it from cache"
-        @handlers[grid_id].delete(handler)
       rescue RpcServer::Error => exc
-        send_message(ws_client, [1, msg_id, {code: exc.code, message: exc.message, backtrace: exc.backtrace}, nil])
+        send_message(ws_client, [1, msg_id, {code: exc.code, message: exc.message}, nil])
         @handlers[grid_id].delete(handler)
       rescue => exc
         error "#{exc.class.name}: #{exc.message}"
-        debug exc.backtrace.join("\n")
-        send_message(ws_client, [1, msg_id, {code: 500, message: "#{exc.class.name}: #{exc.message}", backtrace: exc.backtrace}, nil])
+        debug exc.backtrace.join("\n") if exc.backtrace
```

Comment:
The #1997 [cancer is spreading](https://github.com/kontena/kontena/issues/1997#issuecomment-291082249) :unamused: